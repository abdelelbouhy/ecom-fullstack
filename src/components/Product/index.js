import React from 'react';
import './index.scss';

const Product = (props) => {
    const {title, image: {path, alt}, description, priceLabel, price} = props;

    return (
        <div className='product'>
            <img src={`dist${path}`} alt={alt} className='product_image' />
            <div className='bestseller'>bestseller</div>
            <h1 className='title'>{title}</h1>
            <div className='product_description'>{description}</div>
            <div className='product_price_wrapper'>
                <span className='product_price_wrapper-label'>{priceLabel}</span>
                <span className='product_price_wrapper-price'>{price.toFixed(2)}</span>
            </div>
            <div className='product_shop_now'>Shop Now</div>
        </div>
    );
};


export default Product;