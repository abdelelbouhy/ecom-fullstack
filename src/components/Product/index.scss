.product {
  .product_image {
    width: 100%;
  }
}

@media screen and (min-device-width: 768px) {
  .product {
    width: 33%;
  }
}

@media screen and (min-width: 768px) {
  .product {
    width: 33%;
  }
}

@media screen and (min-device-width: 480px) and (max-device-width: 768px) {
  .product {
    width: 50%;
  }
}

@media screen and (min-width: 480px) and (max-width: 768px) {
  .product {
    width: 50%;
  }
}



@media screen and (min-device-width: 320px) and (max-device-width: 480px) {
  .product {
    width: 100%;
  }
}

@media screen and (min-width: 320px) and (max-width: 480px) {
  .product {
    width: 100%;
  }
}
