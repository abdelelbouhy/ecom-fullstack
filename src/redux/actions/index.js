export const updateList = (data) => {
	return {
		type: 'UPDATE_MOVIES_LIST',
		data
	};
};

export const updateItem = (data) => {
	return {
		type: 'UPDATE_MOVIE_ITEM',
		data
	};
};