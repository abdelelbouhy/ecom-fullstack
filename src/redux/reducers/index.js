const productsList = (state = {data: []}, action) => {
    switch (action.type) {
        case 'UPDATE_PRODUCTS_LIST':
            return Object.assign({}, state, {
                data: action.payload
            });
        default:
            return state;
    }
};


const reducers = {
    productsList
};

export default reducers;
