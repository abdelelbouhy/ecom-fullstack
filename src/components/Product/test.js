import React from 'react';
import {shallow} from 'enzyme';
import Product from './';


describe('Product Component', () => {
    it('should render', () => {
        const props = {
            title: 'Test Title',
            image : {
                path: '/test.jpg',
                alt: 'Alt'
            },
            description: 'Description',
            priceLabel: 'PriceLabel',
            price: 1200
        };
        const wrapper = shallow(<Product {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
