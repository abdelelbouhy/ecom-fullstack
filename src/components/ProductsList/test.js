import React from 'react';
import {mount} from 'enzyme';
import Product from './';
import {createStore, combineReducers} from 'redux';
import {Provider} from 'react-redux';
import {routerReducer} from 'react-router-redux';
import reducers from '../../redux/reducers';


const store = createStore(
    combineReducers({
        ...reducers,
        routing: routerReducer
    })
);

describe('Products List Component', () => {
    it('should render', () => {
        const props = [{
            title: 'Test Title',
            image : {
                path: '/test.jpg',
                alt: 'Alt'
            },
            description: 'Description',
            priceLabel: 'PriceLabel',
            price: 1200
        }];
        const wrapper = mount(<Provider store={store}><Product {...props} /></Provider>);

        expect(wrapper).toMatchSnapshot();
    });
});
