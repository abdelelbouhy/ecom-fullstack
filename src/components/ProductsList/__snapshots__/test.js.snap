// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products List Component should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ProductsList)
    0={
      Object {
        "description": "Description",
        "image": Object {
          "alt": "Alt",
          "path": "/test.jpg",
        },
        "price": 1200,
        "priceLabel": "PriceLabel",
        "title": "Test Title",
      }
    }
  >
    <ProductsList
      0={
        Object {
          "description": "Description",
          "image": Object {
            "alt": "Alt",
            "path": "/test.jpg",
          },
          "price": 1200,
          "priceLabel": "PriceLabel",
          "title": "Test Title",
        }
      }
      products={Array []}
      updateProductsList={[Function]}
    >
      <div
        className="products_list_wrapper"
      />
    </ProductsList>
  </Connect(ProductsList)>
</Provider>
`;
